### ISSUE TITLE
rumqttc: why two executors?

### ISSUE THREAD

**Issue by mattfbacon** [bytebeamio/rumqtt#639]
Why does rumqttc contain both tokio and futures executors?

**Comment by henil** [comment]
i am not sure what you mean by two executors. [futures](https://docs.rs/futures/0.3.28/futures/) is not a async runtime like tokio.

**Comment by henil** [comment]
feel free to re-open if its not clear

**Comment by mattfbacon** [comment]
So they are used for different purposes?

**Comment by henil** [comment]
yes `futures` crate only provides abstraction and extensions for working with futures.

**Comment by mattfbacon** [comment]
As well as an executor gated by the enabled-by-default `executor` feature.

**Comment by henil** [comment]
Oh, good catch. I thought that executor was just used by them for internal tested, didn't know its enabled by default.

**Comment by mattfbacon** [comment]
Ah, alright. I can handle this PR. Remind me in a few days if I forget :)
